scalding-commons-macros/src/test/scala/com/twitter/scalding/commons/macros/PlatformTest.scala

```diff
@@ -87,32 +90,80 @@ class PlatformTest extends WordSpec with Matchers with HadoopSharedPlatformTest
   }
 
   "TBase Test" should {
-    "Expected items should match: TestThriftStructure" in {
+    "Expected items should match: TestThriftStructure2" in {
       runCompareTest[TestThriftStructure]
     }
   }
 
   "ThriftStruct Test" should {
 
-    "Expected items should match : TestStruct" in {
-      runCompareTest[TestStruct]
+    "Expected items should match : Internal Serializer / TestStructdd" in {
+
+      runCompareTest[TestStruct](toScroogeInternalOrderedSerialization[TestStruct], implicitly)
+    }
+
+    "Expected items should match : TProtocol / TestStruct" in {
+      implicit def toScroogeInternalOrderedSerialization[T]: OrderedSerialization[T] = macro ScroogeInternalOrderedSerializationImpl[T]
```

Comment:
So, the exact one beside it is resolved explicitly, though these are often recursive, so you need the implicit to do the recursion. I.e. instead of making one big OrderedSerializer when it finds any inner thrift structs it sticks in a implicitly[InnerStruct], so you need this here to resolve that.